xonsh/main.py

```diff
@@ -163,6 +163,11 @@ def premain(argv=None):
         # and positional arguments passed before the path to the script-file are
         # ignored.
         args.args = arguments[file_index+1:]
+    elif not args.args and other and other[0].startswith('-'):
```

Comment:
i have removed my if clause at line 166 in my latest commits. trying for a more robust way to fix this.